core/src/commonMain/kotlin/com/saveourtool/processbuilder/ProcessBuilderConfig.kt

```diff
@@ -0,0 +1,76 @@
+package com.saveourtool.processbuilder
+
+import okio.Path
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.minutes
+
+/**
+ * Configuration file that
+ *
+ * @property childProcessUsername User which should run the child process
+ * @property defaultExecutionTimeout default timeout [Duration], can be overwritten in [ProcessBuilder.exec], 1 min by default
+ * @property defaultWorkingDirectory [Path] to working directory (`cd $defaultWorkingDirectory` is appended to command)
+ *           or [null] if no working dir change is required, can be overwritten in [ProcessBuilder.exec], [null] by default
+ * @property defaultRedirects
+ */
+data class ProcessBuilderConfig internal constructor(
+    var childProcessUsername: String? = null,
+    var defaultExecutionTimeout: Duration = 1.minutes,
```

Comment:
why a default value is part of config? why not just `executionTimeout`, `workingDirectory` and `redirects`?

Reply:
The question is do we want to provide configuration for `ProcessBuilder` in general or for each run of `ProcessBuilder`. The idea was to configure `ProcessBuilder` for the most of runs while creating it and provide an ability to override the configuration in place. What do you think of this?

Reply:
not clear the benefits: you create `ProcessBuilder` with `ProcessBuilderConfig` and then run `exec` where provides all overrides for `ProcessBuilderConfig`.
I don't see a use case to share the instance of `ProcessBuilder`, for me: I would remove `ProcessBuilderConfig` from constructor and move `ProcessBuilderConfig` to method `exec`